runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -72,11 +97,21 @@ protected void shutDown() throws Exception
 		removeKingdomInfobox();
 	}
 
+	@Provides
+	KingdomConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(KingdomConfig.class);
+	}
+
 	@Subscribe
 	public void onVarbitChanged(VarbitChanged event)
 	{
-		favor = client.getVar(Varbits.KINGDOM_FAVOR);
-		coffer = client.getVar(Varbits.KINGDOM_COFFER);
+		if (client.getGameState() == GameState.LOGGED_IN && throneOfMiscellaniaCompleted() && isInKingdom())
+		{
+			config.setLastChanged(Instant.now());
```

Comment:
this isn't always necessarily the kingdom vars changing here, is that fine?

Reply:
When the `isInKingdom()` guard returns true, the majority of `VarbitChanged` events changing are those of the kingdom values, and the few extraneous events don't impact the kingdom values.

Reply:
Right, I just mean it would bump the time. That's fine?

Reply:
The value really is `lastTimeKnown`. For most cases it shouldn't matter, but I'll add a check to make the value make accurate to its purpose.